src/bin/josh-filter.rs

```diff
@@ -113,11 +124,53 @@ fn run_filter(args: Vec<String>) -> josh::JoshResult<i32> {
         josh::apply_filter_to_refs(
             &repo,
             &*viewobj,
-            &[(src.clone(), t)],
+            &[(src.clone(), t.clone())],
             &mut fm,
             &mut backward_maps.write().unwrap(),
         );
 
+        let mut all_dirs = vec![];
+
+        if check_permissions {
+            let result_tree = repo.find_reference(&t)?.peel_to_tree()?;
+
+            result_tree.walk(git2::TreeWalkMode::PreOrder, |_, entry| {
+                let name = entry.name().unwrap();
+                if name.starts_with("JOSH_ORIG_PATH_") {
+                    let dirname = format!(
+                        "{}",
+                        josh::from_ns(&name.replacen("JOSH_ORIG_PATH_", "", 1))
+                    );
+                    all_dirs.push(dirname);
+                }
+                git2::TreeWalkResult::Ok
+            })?;
+        }
+
+        let mut dedup = vec![];
+
+        for w in all_dirs.as_slice().windows(2) {
```

Comment:
Why is all_dirs already sorted here?

Reply:
It has been built during git2::Tree.walk(). And tree walking happens in order.